client/dashboard/sites/settings-site-visibility/agency-development-site-launch-modal.tsx

```diff
@@ -0,0 +1,68 @@
+import { useMutation } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Button,
+	Modal,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { launchSiteMutation } from '../../app/queries';
+import type { Site } from '../../data/types';
+
+interface AgencyDevelopmentSiteLaunchModalProps {
+	site: Site;
+	onClose: () => void;
+}
+
+export default function AgencyDevelopmentSiteLaunchModal( {
+	site,
+	onClose,
+}: AgencyDevelopmentSiteLaunchModalProps ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const [ isLaunching, setIsLaunching ] = useState( false );
+	const mutation = useMutation( launchSiteMutation( site.slug ) );
+
+	const handleLaunch = () => {
+		setIsLaunching( true );
+		mutation.mutate( undefined, {
+			onSuccess: () => {
+				createSuccessNotice(
+					__( 'Your site has been launched; now you can share it with the world!' ),
+					{
+						type: 'snackbar',
+					}
+				);
+			},
+			onError: ( error: Error ) => {
+				createErrorNotice( error.message || __( 'Failed to launch site' ), {
+					type: 'snackbar',
+				} );
+			},
+			onSettled: () => {
+				setIsLaunching( false );
+				onClose();
+			},
+		} );
+	};
+
+	return (
+		<Modal title={ __( "You're about to launch this website" ) } onRequestClose={ onClose }>
```

Comment:
I don’t think we should use Recoleta for modal titles. Is that specific to this instance or is it a blanket rule we have across WordPress.com?

Reply:
This is a general override that v2 has for `h1`. The Modal component also uses `h1` with no way to customize it. We might need to add a global override.